misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -526,7 +539,36 @@ private StackFrame toDapStackFrame(StackFrameProxyImpl stackFrame) {
             dapStackFrame.setSource(source);
             dapStackFrame.setLine((long) stackFrame.location().lineNumber());
             dapStackFrame.setColumn(0L);
-            dapStackFrame.setName(stackFrame.location().method().name());
+
+            if (stackFrame.location().method().name().matches(WORKER_LAMBDA_REGEX)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) == null) {
+                Value strand = ((ArrayReference) stackFrame.getStackFrame().getArgumentValues().get(0))
```

Comment:
Lets do the `ObjectReference` casting in here, to avoid the other castings in multiple places

Reply:
Done